src/ekore/anomalous_dimensions/unpolarized/space_like/as2.py

```diff
@@ -70,21 +71,23 @@ def gamma_nsp(n, nf, sx):
         Mellin moment
     nf : int
         Number of active flavors
-    sx : numpy.ndarray
-        List of harmonic sums: :math:`S_{1},S_{2}`
+    cache: numpy.ndarray
+        Harmonic sum cache
 
     Returns
     -------
-    gamma_nsp : complex
+    complex
         |NLO| singlet-like non-singlet anomalous dimension
         :math:`\\gamma_{ns,+}^{(1)}(N)`
+
     """
-    S1 = sx[0]
-    S2 = sx[1]
-    Sp1p = harmonics.S1(n / 2)
-    Sp2p = harmonics.S2(n / 2)
-    Sp3p = harmonics.S3(n / 2)
-    g3n = harmonics.g_functions.mellin_g3(n, S1)
+    S1 = c.get(c.S1, cache, n)
+    S2 = c.get(c.S2, cache, n)
+    Sp1p = c.get(c.S1h, cache, n)
+    Sp2p = c.get(c.S2h, cache, n)
+    Sp3p = c.get(c.S3h, cache, n)
+    # TODO: check is this True or False ?
+    g3n = c.get(c.g3, cache, n, is_singlet=False)
```

Comment:
@felixhekhorn Do we want to still compute something at the level of `__init__` ?? Otherwise we will need to pass an additional
`is_singlet` argument around...